### PR TITLE
cmake: thrift requires boost headers, include them as Boost_INCLUDE_DIRS

### PR DESCRIPTION
**Pull request by ideepika**
Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

Fixes # (issue)

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by lalitb** [comment]
Why do we need to bring FindBoost.cmake? Shouldn't this be coming by default with cmake installation:
```
/usr/lib$ dpkg -L cmake-data | grep -i boost
/usr/share/cmake-3.16/Help/module/FindBoost.rst

```

**Comment by ideepika** [comment]
> Why do we need to bring FindBoost.cmake? Shouldn't this be coming by default with cmake installation:
> 
> ```
> $ dpkg -L cmake-data | grep -i boost
> /usr/share/cmake-3.16/Help/module/FindBoost.rst
> /usr/share/cmake-3.16/Modules/FindBoost.cmake
> $
> ```

Indeed this should be packaged with cmake, I will test and update the PR soon, thanks for the review!

**Comment by ideepika** [comment]
@lalitb amended, thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -239,7 +239,10 @@ endfunction()
 
 if(WITH_JAEGER)
   find_package(Thrift QUIET)
-  if(NOT Thrift_FOUND)
+  if(Thrift_FOUND)
+    find_package(Boost REQUIRED)
+    include_directories(${Boost_INCLUDE_DIRS})
```
Comment (lalitb) [inline CMakeLists.txt:244]: @ideepika  - Sorry for replying late on this. Wondering why do we need to explicitly add the Boost headers? Jaeger build is successful in our CI pipeline even without this.

Comment (lalitb) [reply CMakeLists.txt:244]: @ideepika  - Do you have any updates here ? Thanks.

Comment (ideepika) [reply CMakeLists.txt:244]: @lalitb sorry I must have missed the notification, I will check and post update tomorrow, thanks!

**Comment by ideepika** [comment]
In jaeger we had thrift headers which needed boost headers and since we were using external built boost, we needed to define path for Boost Headers in cmake, that should not be an issue when using system installed boost package.
Jenkins tests (in progress) will let us know if the thrift headers for opentelemetry would need the same or not:
https://github.com/ceph/jaeger-client-cpp/commit/7ad60f7330a421fe2adcab1a2d3dbf4159af6bc2 ; will update soon.

**Comment by esigo** [comment]
> In jaeger we had thrift headers which needed boost headers and since we were using external built boost, we needed to define path for Boost Headers in cmake, that should not be an issue when using system installed boost package. Jenkins tests (in progress) will let us know if the thrift headers for opentelemetry would need the same or not: [ceph/jaeger-client-cpp@7ad60f7](https://github.com/ceph/jaeger-client-cpp/commit/7ad60f7330a421fe2adcab1a2d3dbf4159af6bc2) ; will update soon.

@ideepika would it be possible to try setting `CPLUS_INCLUDE_PATH` to where your `boost` is and without changes on `cmake`? Thanks

**Comment by lalitb** [comment]
> In jaeger we had thrift headers which needed boost headers and since we were using external built boost, we needed to define path for Boost Headers in cmake, that should not be an issue when using system installed boost package.
> Jenkins tests (in progress) will let us know if the thrift headers for opentelemetry would need the same or not:
> [ceph/jaeger-client-cpp@7ad60f7](https://github.com/ceph/jaeger-client-cpp/commit/7ad60f7330a421fe2adcab1a2d3dbf4159af6bc2) ; will update soon.

thanks will wait for the update. I thought boost is only the build-time dependency for thrift, and not included in their API headers. Probably it's not the case.

**Comment by ideepika** [comment]
I did some testing:
* when we use self compiled boost library (in case of what we do in @Ceph) there is build failure 
```
ged1209ac/src/test/librados_test_stub/TestIoCtxImpl.cc
CMake Error at src/opentelemetry-cpp-stamp/opentelemetry-cpp-build-None.cmake:49 (message):
  Command failed: 2

   'make' 'opentelemetry_trace' 'opentelemetry_exporter_jaeger_trace'

  See also

    /build/ceph-17.0.0-10252-ged1209ac/obj-x86_64-linux-gnu/src/opentelemetry-cpp/src/opentelemetry-cpp-stamp/opentelemetry-cpp-build-*.log


/usr/bin/ranlib ../../../lib/libceph_immutable_object_cache_lib.a
make[4]: *** [src/CMakeFiles/opentelemetry-cpp.dir/build.make:116: src/opentelemetry-cpp/src/opentelemetry-cpp-stamp/opentelemetry-cpp-build] Error 1
make[4]: Leaving directory '/build/ceph-17.0.0-10252-ged1209ac/obj-x86_64-linux-gnu'
make[3]: *** [CMakeFiles/Makefile2:4594: src/CMakeFiles/opentelemetry-cpp.dir/all] Error 2
```
https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=focal,DIST=focal,MACHINE_SIZE=gigantic/60216//consoleFull

https://shaman.ceph.com/builds/ceph/wip-test-deepika-jaeger/f7c3fcb7357b247541e5e319c7db426cada91ea0/jaeger/293825/
with boost include dirs specified compilation is successful, one thing to note is for opentelemetry CI we install boost-all-dev from package available, that's why I suppose we do face this problem

**Review by lalitb — APPROVED** [review]

**Review by esigo — APPROVED** [review]
Thanks for the PR. LGTM

**Pull request merged**